marketpulse/main/views.py

```diff
@@ -234,3 +234,17 @@ def new_media(request):
                    'location_form': location_form,
                    'mapbox_id': settings.MAPBOX_MAP_ID,
                    'mapbox_token': settings.MAPBOX_TOKEN})
+
+
+def all_media(request):
+    """View gallery with all photo contributions"""
+
+    contributions = Contribution.objects.filter(activity__name=FFXOS_MEDIA_ACTIVITY_NAME)
+
+    # Make sure we have all thumbnails ready
+    for contribution in contributions:
```

Comment:
You could rewrite this to:

``for contribution in Contribution.objects.filter(image_thumb__isnull=True)``

Is null might not work, but you get the idea.